OmsAgent/omsagent.py

```diff
@@ -369,7 +369,7 @@ def stop_telemetry_process():
             for pid in f.readlines():
                 # Verify the pid actually belongs to omsagent.
                 cmd_file = os.path.join("/proc", str(pid.strip("\n")), "cmdline")
-                if os.path.exists(cmd_file):
+                if os.path.exists(cmd_file) and (os.path.getsize(cmd_file) > 0):
                     with open(cmd_file, "r") as pidf:
                         cmdline = pidf.readlines()
                         if cmdline[0].find("omsagent.py") >= 0 and cmdline[0].find("-telemetry") >= 0:
```

Comment:
Here we start our next time of check to time of use issue...

We validate here that the command line includes `omsagent.py` and `-telemetry`...

then on line 377 we send the kill command.

Between the time opening the file, checking that it belongs to `omsagent.py` the `omsagent` could have already been stopped, and the PID re-used by another process. The kill command will at that point send a signal to an innocent process.

Instead this should likely be managed using systemd or some other service manager that can keep track of PID's using cgroups.